File: fixtures/scale/csharp/small/Render/Renderer.cs
using System.Collections.Generic;
using TextPrint.Cli;
using TextPrint.Style;

namespace TextPrint.Render
{
    public class Renderer
    {
        private readonly Options _options;
        private readonly Theme _theme;

        public Renderer(Options options)
        {
            _options = options;
            _theme = Theme.Named(options.ThemeName);
        }

        public string Render(string text)
        {
            var lines = Wrapper.Wrap(text, _options.Width);
            var aligned = new List<string>();
            foreach (var line in lines)
            {
                aligned.Add(Alignment.AlignLine(line, _options.Width, _options.Align));
            }
            if (_options.BorderStyle != "none")
            {
                aligned = new Border(_options.BorderStyle).Frame(aligned, _options.Width);
            }
            return _theme.Apply(string.Join("\n", aligned));
        }
    }
}

File: fixtures/scale/php/small/src/Render/Renderer.php
<?php

namespace TextPrint\Render;

require_once __DIR__ . '/Wrapper.php';
require_once __DIR__ . '/Alignment.php';
require_once __DIR__ . '/Border.php';
require_once __DIR__ . '/../Style/Theme.php';
require_once __DIR__ . '/../Cli/Options.php';

use TextPrint\Cli\Options;
use TextPrint\Style\Theme;

class Renderer
{
    private Options $options;
    private Theme $theme;

    public function __construct(Options $options)
    {
        $this->options = $options;
        $this->theme = Theme::named($options->themeName);
    }

    public function render(string $text): string
    {
        $lines = Wrapper::wrap($text, $this->options->width);
        $aligned = array_map(fn ($line) => Alignment::alignLine($line, $this->options->width, $this->options->align), $lines);
        if ($this->options->borderStyle !== 'none') {
            $aligned = (new Border($this->options->borderStyle))->frame($aligned, $this->options->width);
        }
        return $this->theme->apply(implode("\n", $aligned));
    }
}
